### ISSUE TITLE
Unable to cross compile for linux-arm64 from linux-x86_64

### ISSUE THREAD

**Issue by the123saurav** [xerial/sqlite-jdbc#936]
**Describe the bug**
Unable to cross compile for linux-arm64 from linux-x86_64 when running make native-all
Error is:

```
make: /usr/xcc/aarch64-unknown-linux-gnueabi/bin/aarch64-unknown-linux-gnueabi-gcc: No such file or directory
make: *** [Makefile:65: target/sqlite-3.36.0-Linux-aarch64/sqlite3.o] Error 127
make: *** [linux-arm64] Error 2
```

So it seems its not able to find the compiler in base image?

If I change CROSS_PREFIX to `/usr/xcc/aarch64-unknown-linux-gnu/bin/aarch64-unknown-linux-gnu-` it works.
I also see that the docker image only has `/usr/xcc/aarch64-unknown-linux-gnu/bin/aarch64-unknown-linux-gnu-gcc` file.
OR do we need to install some toolchain before running make in the dockcross command?

**To Reproduce**
I could do this always on my linux AL2 box

**Expected behavior**
Being able to compile for arm.
Also whats the difference between:
`/usr/xcc/aarch64-unknown-linux-gnu/bin/aarch64-unknown-linux-gnu-`
`/usr/xcc/aarch64-unknown-linux-gnueabi/bin/aarch64-unknown-linux-gnueabi-`

**Logs**
```
make: /usr/xcc/aarch64-unknown-linux-gnueabi/bin/aarch64-unknown-linux-gnueabi-gcc: No such file or directory
make: *** [Makefile:65: target/sqlite-3.36.0-Linux-aarch64/sqlite3.o] Error 127
make: *** [linux-arm64] Error 2
```

**Environment (please complete the following information):**
 - OS: Linux
 - CPU architecture: x86_64
 - sqlite-jdbc version 3.36.0.3

**Additional context**

**Comment by the123saurav** [comment]
Hmm also see this https://github.com/xerial/sqlite-jdbc/commit/0ecb95b23f3635bb84a4a535649cf2f8d5f31013
So should I update the makefil natively(i cant update the sqlite version) and port in above changes?

**Comment by gotson** [comment]
Which commit are you based on? The latest master or something else?

**Comment by the123saurav** [comment]
As mentioned in the issue, its 3.36.0.3

**Comment by gotson** [comment]
> As mentioned in the issue, its 3.36.0.3

but that doesn't tell me if you are on an older commit, or if you changed the version.

**Comment by gotson** [comment]
I don't quite understand what you are trying to do, since we have been building the arm64 binaries for quite some time, and it is bundled in the jar.

If you want to build older versions of the libraries and driver, you have 2 options:
1. change the version: https://github.com/xerial/sqlite-jdbc/blob/master/CONTRIBUTING.md#version-update but you may face issues where the code is using more recent features than the one you are trying to build. We don't support building older versions, so you have to fix the problems yourself.
2. checkout the commit you want, and build from there. It seems that is what you are currently doing. We also don't support building older versions with that method.

The questions you have are dockcross specific, you would better ask on their github directly. We use dockcross, but we are not expert in it.

**Comment by the123saurav** [comment]
Hi @gotson , yeah 2nd is what I am trying to do as i have to compile with some flags.

I also stumbled upon https://github.com/xerial/sqlite-jdbc/commit/0ecb95b23f3635bb84a4a535649cf2f8d5f31013 where we seem to update the path for compiler.

All I want to know at this point is(I am not a C++ guy hence bit of confusion) that its safe to make this change in Makefile in my current version(3.36.0.3 release of Xerial JDBC lib)?
I was able to successfully compile but still wanted to ask here specifically.
Hope this info helps :)
